src/core/net/ip6.cpp

```diff
@@ -1035,8 +1036,15 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
             Udp::Header udp;
 
             IgnoreError(aMessage.Read(aMessage.GetOffset(), udp));
-            VerifyOrExit(Get<Udp>().ShouldUsePlatformUdp(udp.GetDestinationPort()), error = kErrorNoRoute);
 
+            if(mIsReceiveIp6FullFilterEnabled)
+            {
+                VerifyOrExit(Get<Udp>().IsPortInUse(udp.GetDestinationPort()) == false, error = kErrorNoRoute);
```

Comment:
In the case of `PLAT_UDP + full filter`, the packet won't be passed to the Thread netif (e.g. wpan0) if the port is bound in OT. 
However, currently OT (in Udp::HandleMessage) only pass the message to `HandlePayload` if it's not using Platform UDP (e.g. MLE, TMF). 
We might need to update `Udp::HandleMessage` too accordingly.

I think we need to add test for that. For example, we need to run `test_plat_udp_accessibility.py` using `full filtering`.

Reply:
Could you explain this some more please? How is HandleMessage related to my changes? Where is this test_plat_udp_accessibility.py test located?